packages/shared/src/components/modals/post/CommentModal.tsx

```diff
@@ -0,0 +1,262 @@
+import React, {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { QueryClient, useQueryClient } from '@tanstack/react-query';
+import classNames from 'classnames';
+import { Modal, LazyModalCommonProps } from '../common/Modal';
+import { FormWrapper } from '../../fields/form';
+import {
+  CommentMarkdownInput,
+  CommentMarkdownInputProps,
+} from '../../fields/MarkdownInput/CommentMarkdownInput';
+import { useMutateComment } from '../../../hooks/post/useMutateComment';
+import { useVisualViewport } from '../../../hooks/utils/useVisualViewport';
+import { RequestKey, generateQueryKey } from '../../../lib/query';
+import { Comment, PostCommentsData } from '../../../graphql/comments';
+import { useNotificationToggle } from '../../../hooks/notifications';
+import { NotificationPromptSource } from '../../../lib/analytics';
+import { Switch } from '../../fields/Switch';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import CommentContainer from '../../comments/CommentContainer';
+
+const getCommentFromCache = ({
+  client,
+  postId,
+  commentId,
+}: {
+  client: QueryClient;
+  postId: string;
+  commentId?: string;
+}): Comment | undefined => {
+  if (!commentId) {
+    return undefined;
+  }
+
+  const queryKey = generateQueryKey(RequestKey.PostComments, null, postId);
+
+  const data = client.getQueryData<PostCommentsData>(queryKey);
+
+  if (!data) {
+    return undefined;
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const item of data?.postComments?.edges) {
+    if (item.node.id === commentId) {
+      return item.node;
+    }
+    if (item.node.children) {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const child of item.node.children.edges) {
+        if (child.node.id === commentId) {
+          return child.node;
+        }
+      }
+    }
+  }
+
+  return undefined;
+};
+
+// statically defined heights of various containers that
+// affect final height of the input container
+// when styles change, these should be updated accordingly
+const headerSize = 57;
+const replySize = 40;
+const footerSize = 48;
```

Comment:
This looks highly bug prone.
Can't we get them from calculations in JS rather if you need them there?

Maybe by using: `globalThis?.window?.getComputedStyle`

Reply:
Refactored to get the proper sizes from element refs instead.